### PR TITLE
Ltm 120 score normal

### PR DESCRIPTION
**Pull request by FKGSOFTWARE**
Normalising score functionality added in the interface, helper function at the top of the file for use in different classes. Successfully tested with migr-reeval.py

### CODE REVIEW

File: dev_scripts/migr-reeval.py [hunk 1]
```diff
@@ -21,8 +21,18 @@ def cost_callback(cost_usd: float):
...
         result = TestResult.from_file(path)
-        result.score, result.max_score, result.reasoning = ds.evaluate_correct(
-            questions=[],  # They shouldn't be necessary for the evaluation
+        # result.score, result.max_score, result.reasoning = ds.evaluate_correct(
+        # Original line commented out for clarity:
+        # result.score, result.max_score, result.reasoning = ds.evaluate_correct(
+        #     questions=[],  # They shouldn't be necessary for the evaluation
+        #     responses=result.actual_responses,
+        #     expected_answers=result.expected_responses,
+        # )
```
Comment (JosephDavidsonKSWH) [inline dev_scripts/migr-reeval.py:30]: You can remove this commented stuff

File: dataset_interfaces/interface.py [hunk 2]
```diff
@@ -119,8 +124,16 @@ def reset_message(self) -> str:
         return self.dataset_generator.reset_message
 
     @property
-    def evaluation_fn(self) -> Callable[[List[str], list[str], List[Any]], tuple[int, int, List[str]]]:
-        return self.dataset_generator.evaluate_correct
+    # def evaluation_fn(self) -> Callable[[List[str], list[str], List[Any]], tuple[int, int, List[str]]]:
+    #     return self.dataset_generator.evaluate_correct
```
Comment (JosephDavidsonKSWH) [inline dataset_interfaces/interface.py:128]: Can be removed

File: dataset_interfaces/interface.py [hunk 3]
```diff
@@ -294,6 +307,11 @@ def evaluate_correct(
         self, questions: List[str], responses: List[str], expected_answers: List[Any]
     ) -> Tuple[int, int, List[str]]:
         pass
+    
+    def evaluation_fn(self) -> Callable[[List[str], List[str], List[Any]], Tuple[float, int, List[str]]]:
+        def evaluator(questions: List[str], responses: List[str], expected_answers: List[Any]) -> Tuple[float, int, List[str]]:
+            return normalize_scores(self.evaluate_correct, questions, responses, expected_answers)
+        return evaluator
```
Comment (JosephDavidsonKSWH) [inline dataset_interfaces/interface.py:314]: Why is this here? The `evaluation_fn` is called as a part of the `TestExample`, not `DatasetInterface`

**Comment by JosephDavidsonKSWH** [comment]
This has a merge conflict in the `spy_meeting` dataset file. You can rebase these changes on `dev` and resolve the conflict in favour of `devs` changes (the changes you made are already in `dev`)

File: dataset_interfaces/interface.py (hunk 3 above)
Comment (FKGSOFTWARE) [reply dataset_interfaces/interface.py:312]: This was added as the evaluation in the `migr-reeval.py` file uses the `DatasetInterface` class not the `TestExample` class

**Review by JosephDavidsonKSWH — APPROVED** [review]
